src/features/diagnosticsProvider.ts

```diff
@@ -280,41 +275,39 @@ class DiagnosticsProvider extends AbstractSupport {
             .filter(diagnosticInFile => diagnosticInFile !== undefined);
     }
 
-    private _validateEntireWorkspace(): NodeJS.Timeout {
-        return setTimeout(async () => {
-            let value = await serverUtils.codeCheck(this._server, { FileName: null }, new vscode.CancellationTokenSource().token);
+    private async _validateEntireWorkspace() {
+        let value = await serverUtils.codeCheck(this._server, { FileName: null }, new vscode.CancellationTokenSource().token);
```

Comment:
No need to use set Timeouts the debounce has already waited to execute this code.